setup-ubuntu-vm-gpu.sh

```diff
@@ -0,0 +1,129 @@
+	if [ $dc = 0 ]; then
+		i=0
+		len=${#msg}
+		len=$[$len+$maxDots]
+		b=""
+		while [ $i -ne $len ]
+		do
+			b="$b "
+			i=$[$i+1]
+		done
+		echo -e -n "\r$b"
+		dc=1
+	else
+		msg="$msg"
+		i=0
+		while [ $i -ne $dc ]
+		do
+			msg="$msg."
+			i=$[$i+1]
+		done
+		dc=$[$dc+1]
+		if [ $dc = $maxDots ]; then
+			dc=0
+		fi
+	fi
+	export dotCount=$dc
+	echo -e -n "\r$msg"
+}
+
+function serviceStatusCheck() {
+    # This function checks service is active or inactive.
+    timeCheck=0
+    while true
+      do
+        service=$1
+        exit_required=$2
+        DOCKER_SERVICE_STATUS="$(systemctl is-active $service)"
+        if [ "${DOCKER_SERVICE_STATUS}" = "active" ]; then
+          echo ""
+          echo "$service running.."
+          break
+        fi
+        showMessage "$service status check"
+        sleep $SLEEP_INTERVAL
+        timeCheck=$[timeCheck+$SLEEP_INTERVAL]
+        if [ $timeCheck -gt $TIMEOUT ]; then
+          echo ""
+          echo "$service not running, Timeout error."
+          echo ""
+          if [ "${exit_required}" = "True" ]; then
+            exit 1
+          fi
+        fi
+      done
+}
+
+echo "1. Set nvidia docker runtime:"
+cat <<EOF >/etc/docker/daemon.json
+{
+    "default-runtime": "nvidia",
+    "runtimes": {
+        "nvidia": {
+            "path": "/usr/bin/nvidia-container-runtime",
+            "runtimeArgs": []
+        }
+    }
+}
+EOF
+systemctl restart docker
+serviceStatusCheck "docker.service" "True"
+
+echo "2. Install kubeadm, kubectl and kubelet:"
+apt-get update -y && apt-get install -y openssh-server apt-transport-https curl
+curl -s https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+cat <<EOF >/etc/apt/sources.list.d/kubernetes.list
+deb http://apt.kubernetes.io/ kubernetes-xenial main
+EOF
+apt-get update -y && apt-get install -y kubelet=1.21.0-00 kubeadm=1.21.0-00 kubectl=1.21.0-00
+systemctl daemon-reload && systemctl start kubelet && systemctl enable kubelet && systemctl status kubelet
+serviceStatusCheck "kubelet.service" "False"
+
+echo "3. Setup helm"
+curl -L -O https://get.helm.sh/helm-v3.3.4-linux-amd64.tar.gz && tar -xvf helm-v3.3.4-linux-amd64.tar.gz && mv linux-amd64/helm /usr/local/bin/ && rm helm-v3.3.4-linux-amd64.tar.gz
+helm version
+
+echo "4. Initialize kubernetes cluster:"
+kubeadm init
+mkdir -p $HOME/.kube && cp -i /etc/kubernetes/admin.conf $HOME/.kube/config && chown $(id -u):$(id -g) $HOME/.kube/config
+
+echo "5. Install network driver:"
+kubectl apply -f https://docs.projectcalico.org/v3.9/manifests/calico-typha.yaml
+kubectl -n kube-system set env daemonset/calico-node FELIX_IGNORELOOSERPF=true
+
+echo "6. Remove NoSchedule taint from master:"
+kubectl taint nodes $(kubectl get nodes --selector=node-role.kubernetes.io/master | awk 'FNR==2{print $1}') node-role.kubernetes.io/master-
+
+echo "7. Setup gpu device driver:"
+kubectl apply -f https://raw.githubusercontent.com/NVIDIA/k8s-device-plugin/1.0.0-beta4/nvidia-device-plugin.yml
```

Comment:
This script is almost the same as setup-ubuntu-vm.sh except for this line and installing the nvidia container runtime. We should consider breaking both of those scripts into functions and import them so we can avoid the amount of copy/paste code.